libs/vm-modules/tests/unit/math/tensor_tests.cpp

```diff
@@ -355,6 +357,69 @@ TEST_F(MathTensorTests, tensor_set_from_string)
   EXPECT_TRUE(gt.AllClose(tensor->GetTensor()));
 }
 
+TEST_F(MathTensorTests, tensor_shape_from_tensor)
+{
+  static char const *tensor_from_string_src = R"(
+    function main() : Array<UInt64>
+      var tensor_shape = Array<UInt64>(3);
+      tensor_shape[0] = 2u64;
+      tensor_shape[1] = 3u64;
+      tensor_shape[2] = 4u64;
+      var x = Tensor(tensor_shape);
+
+      var shape = x.shape();
+
+      return shape;
+
+    endfunction
+  )";
+
+  ASSERT_TRUE(toolkit.Compile(tensor_from_string_src));
+  Variant res;
+  ASSERT_TRUE(toolkit.Run(&res));
+
+  auto tensor_shape{res.Get<Ptr<IArray>>()};
+
+  std::vector<SizeType> ret;
+  std::vector<SizeType> gt({2, 3, 4});
+
+  while (tensor_shape->Count() > 0)
+  {
+    ret.emplace_back(tensor_shape->PopFrontOne().Get<uint64_t>());
+  }
+
+  EXPECT_TRUE(gt == ret);
```

Comment:
EXPECT_EQ (if fails) returns both arguments' names and their values, while EXPECT_TRUE just notifies that "it is false", so I would prefer  
 EXPECT_EQ(gt, ret);